daemon/cluster/executor/container/controller.go

```diff
@@ -155,7 +155,11 @@ func (r *controller) Prepare(ctx context.Context) error {
 			return exec.ErrTaskPrepared
 		}
 
-		return err
+		if r.pullErr != nil {
+			return errors.Wrap(r.pullErr, err.Error())
+		} else {
```

Comment:
the `else` is not needed here; just

```go
if r.pullErr != nil {
    return errors.Wrap(r.pullErr, err.Error())
}
return err
```

Reply:
@thaJeztah Thanks! I updated(and squashed).